src/world/terrain/material.cpp

```diff
@@ -59,4 +57,110 @@ TerrainColorMapping::assign_color_mapping(
     }
 }
 
+MaterialGroup::MaterialGroup(const std::vector<generation::material_designation_t>& data
+) :
+    contain_all_materials(false) {
+    // want to return a group that represents the given data
+    // There will be elements with no requirements on the color
+    //    std::set<MaterialId> materials;
+    // amd some with requirements on the color
+    //    std::map<MaterialId, std::set<ColorId>> materials_w_color;
+
+    for (const generation::material_designation_t& material_data : data) {
+        // read the material id from the data
+        // MaterialId mat_id = material_data.material;
+
+        const auto& material = material_data.material;
+        const auto& color = material_data.color;
+
+        std::unordered_set<ColorId> colors;
+        std::unordered_set<MaterialId> materials;
+
+        {
+            switch (material.index()) {
```

Comment:
NO NO NO NO NO NO NO NO NO this is UB

Use a visistor: https://en.cppreference.com/w/cpp/utility/variant/visit